<launch>
  <!-- Gen3 specific arguments required by Kortex API for instantiating a private communication session -->
  <arg name="ip_address" default="emprise-kortex.coecis.cornell.edu" doc="IP address for TCP/UDP comms"/>
  <arg name="username" default="admin" doc="Username for kortex session"/>
  <arg name="password" default="admin" doc="Password for kortex session"/>
  <arg name="api_session_inactivity_timeout_ms" default="35000"/> <!--milliseconds-->
  <arg name="api_connection_inactivity_timeout_ms" default="20000"/> <!--milliseconds-->
  
  <!-- Gen3 Hardware Node -->
  <node name="kortex_hardware" pkg="kortex_hardware" type="kortex_hardware"
          output="screen">
      <param name="ip_address" value="$(arg ip_address)"/>
      <param name="username" value="$(arg username)"/>
      <param name="password" value="$(arg password)"/>
      <param name="api_session_inactivity_timeout_ms" value="$(arg api_session_inactivity_timeout_ms)"/>
      <param name="api_connection_inactivity_timeout_ms" value="$(arg api_connection_inactivity_timeout_ms)"/>
  </node>

  <!-- start joint state and mode controllers -->
  <node name="controller_spawner_started" pkg="controller_manager" type="spawner" respawn="false"
      output="screen"
      args="
        joint_state_controller
        joint_mode_controller
      " />

  <!-- load controller configuration -->
  <rosparam file="$(find libada)/config/gen3_$(arg dof)dof_controllers.yaml" command="load" />

  <!-- Default Controllers -->
  <node name="base_controller_spawner_stopped" pkg="controller_manager" type="spawner" respawn="false"
      output="screen"
      args="
        --stopped
        velocity_controller
        trajectory_controller
        hand_controller
        " />
</launch>